core-web/libs/portlets/dot-content-drive/portlet/src/lib/components/dot-content-drive-toolbar/components/dot-content-drive-baseType-selector/dot-content-drive-baseType-selector.component.ts

```diff
@@ -0,0 +1,87 @@
+import { of } from 'rxjs';
+
+import { Component, inject, model } from '@angular/core';
+import { FormsModule } from '@angular/forms';
+
+import { CheckboxModule } from 'primeng/checkbox';
+import { MultiSelectModule } from 'primeng/multiselect';
+
+import { debounceTime, distinctUntilChanged, map, take } from 'rxjs/operators';
+
+import { DotContentTypeService } from '@dotcms/data-access';
+import { DotMessagePipe } from '@dotcms/ui';
+
+import { DEBOUNCE_TIME, MAP_NUMBERS_TO_BASE_TYPES } from '../../../../shared/constants';
+import { DotContentDriveStore } from '../../../../store/dot-content-drive.store';
+
+@Component({
+    selector: 'dot-content-drive-base-type-selector',
+    templateUrl: './dot-content-drive-baseType-selector.component.html',
+    styleUrl: './dot-content-drive-baseType-selector.component.scss',
+    imports: [MultiSelectModule, FormsModule, CheckboxModule, DotMessagePipe],
+    standalone: true
+})
+export class DotContentDriveBaseTypeSelectorComponent {
+    $selectedBaseTypes = model<string[]>([]);
+
+    readonly #store = inject(DotContentDriveStore);
+    readonly #dotContentTypeService = inject(DotContentTypeService);
+
+    readonly $state = signalState({
+        baseTypes: []
+    });
+
+    ngOnInit() {
+        this.getCurrentBaseTypes();
+
+        this.#dotContentTypeService
+            .getAllContentTypes()
+            .pipe(
+                take(1),
+                map((response) => response.filter((item) => item.name !== 'FORM'))
+            )
+            .subscribe((response) => {
+                patchState(this.$state, {
+                    baseTypes: response
+                });
+            });
+    }
+
+    getCurrentBaseTypes() {
+        const baseTypes = this.#store.getFilterValue('baseType') as string[];
+
+        if (baseTypes?.length > 0) {
+            const values = baseTypes.map((key) => MAP_NUMBERS_TO_BASE_TYPES[key]).filter(Boolean);
+
+            this.$selectedBaseTypes.set(values);
+        }
+    }
+
+    onChange() {
+        of(this.$selectedBaseTypes() ?? [])
+            .pipe(
+                debounceTime(DEBOUNCE_TIME), // Debounce to avoid spamming the server
+                distinctUntilChanged()
+            )
+            .subscribe((value) => {
+                if (value.length > 0) {
+                    // Get all keys from MAP_NUMBERS_TO_BASE_TYPES where the values match the array
+                    const keys = value
+                        .map(
+                            (val) =>
+                                Object.entries(MAP_NUMBERS_TO_BASE_TYPES).find(
+                                    ([_key, v]) => v === val
+                                )?.[0]
+                        )
+                        .filter(Boolean);
+
+                    this.#store.patchFilters({
+                        baseType: keys
+                    });
+                } else {
+                    this.#store.removeFilter('baseType');
+                }
```

Comment:
Why this is not a dumb component that emit the selected value and outside we do what we have to do?

Reply:
Because we don't want to bloat the Toolbar component, there will be more logic, we still have pending the workflow actions and bulk actions